atom/renderer/atom_renderer_client.cc

```diff
@@ -129,6 +130,9 @@ void AtomRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
   new PepperHelper(render_frame);
   new AtomRenderFrameObserver(render_frame, this);
+
+  // Allow file scheme to handle service worker by default.
+  blink::WebSecurityPolicy::registerURLSchemeAsAllowingServiceWorkers("file");
```

Comment:
Its required to bypass the security restrictions of blink layer, `AddServiceWorkerSchemes` is only used on the browser side by `ServiceWorker` request handler.